### DISCUSSION TITLE
Update checkout block after selected payment method changed

### DISCUSSION THREAD

**Discussion by ColonelHathii** [woocommerce/woocommerce#45002] [WooCommerce, Blocks, and Site Editing]
Hi folks 👋 

We added a fee when our gateway is selected (using the hook "`woocommerce_cart_calculate_fees`").

In the checkout which is inlcuded via shortcode wie triggered when the payment method has changed to force updating the checkout:

```
$( document.body ).on( 'change', 'input[name="payment_method"]', function() {
	$( 'body' ).trigger( 'update_checkout' );
} );
```

How can we achieve this same behaviour within the checkout block?

Thanks in advance.
Cheers 🍻

**Comment by opr** [comment]
Hi, when your gateway is selected it should render its `content` component. Within this component you could run a `useEffect` with an empty deps array, so it runs once on mount. In this effect you can use a feature we have that lets you update the cart on the server following a client-side action. [`extensionCartUpdate`](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/third-party-developers/extensibility/rest-api/extend-rest-api-update-cart.md).

This will cause some server-side processing (that you define) to occur and return the new updated cart.

**Reply by ColonelHathii** [reply]
Hi @opr 👋 

thank you very much for your hint, with these we could implement our goal.

However, we have not implemented it in the content "component", but elsewhere, because otherwise useEffect did not work when switching from our gateway to another. We still had some problems that the fee was displayed and calculated when selecting our gateway **_and when selecting another gateway the fee was removed again_**. With a few tricks it now works with Checkout Block as well as with Checkout Shortcode.

Thanks again.
